ha/core/event_manager/subscribe_event.py

```diff
@@ -31,9 +33,26 @@ def __init__(self, resource_type : RESOURCE_TYPES, states : List[RESOURCE_STATUS
             states (list): States
         """
         self.states = []
+        self.functional_types = []
         self.resource_type = resource_type.value \
             if isinstance(resource_type, RESOURCE_TYPES) \
                 else resource_type
         for state in states:
             st = state.value if isinstance(state, RESOURCE_STATUS) else state
             self.states.append(st)
+
+        if functional_types:
+            try:
+                resource_func_types = FUNCTIONAL_TYPES[self.resource_type.upper()]
+            except KeyError as err:
+                raise Exception(f"Unsupported resource type '{err}'")
+
+            for func_type in functional_types:
+                func_type = func_type.lower()
+                func_type = resource_func_types.value(func_type).value
+                self.functional_types.append(func_type)
+
+        # else:
```

Comment:
functional_types should be a mandatory parameter I think. That way every component which is subscribing knows exactly which all events it is subscribing to 
@shaileshsvc any comments ?

Reply:
Right we can either make it mandatory or specify that in absence of any functional type specification it will be treated as ALL functional types for given resource type.

Reply:
I would like to address this "ALL" functional type in a separate ticket. PLease share your thought. @shaileshsvc @ArchanaLimaye

Reply:
[CORTX-30826](https://jts.seagate.com/browse/CORTX-30826) - Subscription logic to allow ALL as functional type and process all resource types
I created a separate ticket for this.